test/ITablelandController.ts

```diff
@@ -0,0 +1,216 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers"
+import chai from "chai"
+import chaiAsPromised from "chai-as-promised"
+// @ts-ignore-start
+import { ethers } from "hardhat"
+// @ts-ignore-end
+import type {
+  TablelandTables,
+  TestERC721Enumerable,
+  TestERC721AQueryable,
+  TestTablelandController,
+  TestAllowAllTablelandController,
+} from "../typechain-types"
+
+chai.use(chaiAsPromised)
+const expect = chai.expect
+
+describe("ITablelandController", function () {
+  let accounts: SignerWithAddress[]
+  let tables: TablelandTables
+  let foos: TestERC721Enumerable
+  let bars: TestERC721AQueryable
+  let controller: TestTablelandController
+  let allowAllController: TestAllowAllTablelandController
+
+  beforeEach(async function () {
+    accounts = await ethers.getSigners()
+
+    // Deploy tables
+    tables = (await (
+      await ethers.getContractFactory("TablelandTables")
+    ).deploy()) as TablelandTables
+    await tables.deployed()
+    await (await tables.initialize("https://foo.xyz/")).wait()
+
+    // Deploy test erc721 contracts
+    foos = (await (
+      await ethers.getContractFactory("TestERC721Enumerable")
+    ).deploy()) as TestERC721Enumerable
+    await foos.deployed()
+    bars = (await (
+      await ethers.getContractFactory("TestERC721AQueryable")
+    ).deploy()) as TestERC721AQueryable
+    await bars.deployed()
+
+    // Deploy test controllers
+    controller = (await (
+      await ethers.getContractFactory("TestTablelandController")
+    ).deploy()) as TestTablelandController
+    await controller.deployed()
+
+    allowAllController = (await (
+      await ethers.getContractFactory("TestAllowAllTablelandController")
+    ).deploy()) as TestAllowAllTablelandController
+    await allowAllController.deployed()
+
+    // Setup controller
+    await (await controller.setFoos(foos.address)).wait()
+    await (await controller.setBars(bars.address)).wait()
+  })
+
+  it("Should set controller for a table", async function () {
```

Comment:
If I want to "unset" the controller, do I set the Controller to the owner address?
Maybe we should add a test for "unsetting" the controller? By unsetting I mean get the table back to the same access control it would have before a controller is set.

Reply:
oh good call! i'll add a test... you'd just set it to the zero address, but not sure what that looks like yet in TS